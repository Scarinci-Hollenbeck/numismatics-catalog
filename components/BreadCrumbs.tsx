import React from 'react';
import Link from 'next/link';
import { makeTitle } from '../utils/helpers';

type Props = {
  links: Array<string>
}

export default function BreadCrumbs({ links }: Props): JSX.Element {
  return (
    <ul className="breadcrumbs">
      <li>
        <Link href="/">
          <a>Home</a>
        </Link>
      </li>
      {links.map((link) => (
        <li key={link}>
          <Link href={link}>
            <a>{makeTitle(link)}</a>
          </Link>
        </li>
      ))}

      <style jsx>
        {`
          .breadcrumbs {
            font-family: 'Tajawal Regular';
            margin: 0;
            margin-bottom: 1em;
            padding: 0;
          }

          .breadcrumbs li {
            display: inline;
            margin-right: 10px;
          }

          .breadcrumbs li a {
            color: #fff;
          }
        `}
      </style>
    </ul>
  );
}
