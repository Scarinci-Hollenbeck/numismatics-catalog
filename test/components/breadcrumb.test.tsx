import React from 'react';
import renderer from 'react-test-renderer';
import BreadCrumbs from '../../components/BreadCrumbs';

const crumbs = [
  {
    link:
      'http://test.com/coin/5f6e4147ae97b21b48da848e/king-alfonso-xii%2C-1874---1885',
    title: 'Coin Item 1',
  },
  {
    link:
      'http://test.com/coin/5f6e4147ae97b21b48da848e/king-alfonso-xi%2C-1840---1874',
    title: 'Coin Item 2',
  },
];

describe('BreadCrumbs Component', () => {
  it('expect coins length == 2', () => {
    expect(crumbs.length).toBe(2);
  });

  it('The word "Coin" is found in the title', () => {
    const { title } = crumbs[0];
    expect(title).toMatch(/Coin/);
  });

  it('the coin article list contains sourceUrl and title', () => {
    expect(crumbs[1]).toHaveProperty('title', 'Coin Item 2');
    expect(crumbs[1]).toHaveProperty(
      'link',
      'http://test.com/coin/5f6e4147ae97b21b48da848e/king-alfonso-xi%2C-1840---1874',
    );
  });

  it('matches snapshot', () => {
    const tree = renderer.create(<BreadCrumbs crumbs={crumbs} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
