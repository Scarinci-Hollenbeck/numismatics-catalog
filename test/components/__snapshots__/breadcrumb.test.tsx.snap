// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadCrumbs Component matches snapshot 1`] = `
<ul
  className="jsx-2859769289 breadcrumbs"
>
  <li
    className="jsx-2859769289"
  >
    <a
      className="jsx-2859769289"
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      Home
    </a>
  </li>
  <li
    className="jsx-2859769289"
  >
    <a
      className="jsx-2859769289"
      href="http://test.com/coin/5f6e4147ae97b21b48da848e/king-alfonso-xii%2C-1874---1885"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      Coin Item 1
    </a>
  </li>
  <li
    className="jsx-2859769289"
  >
    <a
      className="jsx-2859769289"
      href="http://test.com/coin/5f6e4147ae97b21b48da848e/king-alfonso-xi%2C-1840---1874"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      Coin Item 2
    </a>
  </li>
</ul>
`;
